packages/cdktf/lib/tokens/private/resolve.ts

```diff
@@ -234,15 +263,17 @@ export function resolve(obj: any, options: IResolveOptions): any {
   // mistake somewhere and resolve will get into an infinite loop recursing into
   // child.parent <---> parent.children
   if (isConstruct(obj)) {
-    throw new Error("Trying to resolve() a Construct at " + pathName);
+    throw new Error(`Trying to resolve() a Construct at '${pathName}'. 
+This often means that there is an unintended cyclic dependency in your construct tree.
+This leads to the resolution being stuck in an infinite loop and will eventually fail.`);
```

Comment:
```suggestion
throw new Error(`Trying to resolve() a Construct at '${pathName}'. 
This often means that there is an unintended cyclic dependency in your construct tree, leading to the resolution being stuck in an infinite loop which will eventually fail.`);
```

Another nit- feel free to ignore